### PR TITLE
paging and notifications for badges

### PR DESCRIPTION
**Pull request by xithan**

### CODE REVIEW

File: mod/badges/set/abstract/01_badge.rb [hunk 1]
```diff
@@ -74,11 +89,12 @@ def compare_actions other
 end
 
 def compare_levels other
-  if badge_level == other.badge_level
-    affinity_type == :general ? 1 : -1
-  else
-    badge_level_index <=> other.badge_level_index
-  end
+  badge_level_index <=> other.badge_level_index
```
Comment (ethn) [inline mod/badges/set/abstract/01_badge.rb:129]: nice

File: mod/badges/spec/set/right/badges_earned_spec.rb [hunk 2]
```diff
@@ -62,22 +66,16 @@ def badge_count level=nil
...
-                         "Answer Advancer",
-                         "Commentator"]
+      is_expected.to eq ordered_badges
     end
   end
 
   describe "html format" do
     describe "view :core" do
-      #it
+      # it
```
Comment (ethn) [inline mod/badges/spec/set/right/badges_earned_spec.rb:78]: much better :)

File: mod/badges/spec/set/type_plus_right/wikirate_company/logo/award_badges_spec.rb [hunk 3]
```diff
@@ -9,9 +9,12 @@
     let(:sample_acting_card) { sample_company.field(:logo, new: {}) }
 
     def execute_awarded_action number
+      company = nil
       Card::Auth.as_bot do
         company = Card.create! type_id: Card::WikirateCompanyID,
                                name: "Company #{number}"
+      end
+      Card::Auth.as_bot do
```
Comment (ethn) [inline mod/badges/spec/set/type_plus_right/wikirate_company/logo/award_badges_spec.rb]: what is this about?

**Review by ethn — CHANGES_REQUESTED** [review]
only "requested change" really is an explanation of the `as_bot` code near my last comment.  Otherwise this looks solid to me.

File: mod/badges/spec/set/type_plus_right/wikirate_company/logo/award_badges_spec.rb (hunk 3 above)
Comment (xithan) [reply mod/badges/spec/set/type_plus_right/wikirate_company/logo/award_badges_spec.rb]: it doesn't work in a single `as_bot` block. "no permission to create +*logo"

Comment (xithan) [reply mod/badges/spec/set/type_plus_right/wikirate_company/logo/award_badges_spec.rb]: seems like the `as_bot` status gets lost somewhere

**Pull request merged**